design/005-Wire-Encoding-Abstraction.md

```diff
@@ -0,0 +1,118 @@
+# Proposal: Abstraction of Wire Messages Encoding
+
+* Author(s): Sebastian Stammler (@sebastianst)
+* Affected repository: go-perun
+
+## Summary
+
+The goal is to be able to choose different de/-serialization methods for wire
+messages when using go-perun's `wire` package for inter-node wire message
+exchange. After the changes, it should be possible to choose protobuf
+serialization.
+
+## Motivation
+
+Wire messages in go-perun currently use a home-baked and non-exchangeable
+encoding and decoding format. When developing new Perun channel client
+implementations, it would currently be necessary to mirror this custom
+serialization format.
+
+## Details
+
+Wire messages in go-perun currently use a home-baked and non-exchangeable
+encoding and decoding format, as specified by `Encode` and `Decode`
+methods on wire messages (see, e.g., message definitions in `client/*msgs.go`),
+methods `Envelope.(En|De)code` and functions `(En|De)code` in `wire/msg.go`.
+
+It should be possible to choose a different wire serialization protocol, e.g., protobuf.
+
+### Note on `pkg/io` serialization
+
+Note, thought, that this custom format is _not_ Go-specific. It is a packed,
+little-endian binary encoding format and the serialization logic of actual Go
+types is defined in package `pkg/io` and in particular in file
+`pkg/io/serialize.go`. (However, I just realize that `*big.Int`s are actually
+encoded in big-endian byte order...so we might as well switch to big-endianess
+in `pkg/io/serialize.go` to be more consistent
+([related issue](https://github.com/hyperledger-labs/go-perun/issues/167)).)
+
+## Rationale
+
+See #motivation.
+
+## Impact
+
+This can be implemented in a backwards-compatible fashion, where `pkg/io`
+encoding would be the default and other serialization backends could be injected
+at runtime.
```

Comment:
The objective of this section to describe the impact level in one word. Could you assign one of the four impact levels ? 
I think it could be `New Feature`, because the `pkg/io` will be set as default backend and the usage of go-perun wold remain the same. If required, the `SetEnvelopeSerializer` function can be used to assign a different serialization backend.
```
Minor (Does not impact any existing features)
Major (Breaks one or more existing features)
New Feature (Introduces a functionality)
Architecture (Requires a modification of the architecture) 
```

You could then move this text to impact to Rationale section.

`The suggested implementation is backwards-compatible, where `pkg/io` will be the default and other.... `